### ISSUE TITLE
TestTranslateTimestamp shouldn't use time.Now when testing for days in months

### ISSUE THREAD

**Issue by fabioy** [kubernetes/kubernetes#17998]
Build is "broken" because TestTranslateTimestamp in pkg/kubectl/resource_printer_test.go does a "time.Now().AddDate(0, -3, 0)" and compares it to "92d". Test should either use a fixed date, or allow for variability in days per month.

Failed tests:
1 tests failed.
FAILED:  kubectl.TestTranslateTimestamp

Error Message:
Failed

Stack Trace:
resource_printer_test.go:1226: On months ago, expected '92d', but got '91d'

**Comment by fabioy** [comment]
PR #18000 in flight.

**Comment by karlkfi** [comment]
Did the code that broke this unit test also get cherry-picked? I'm seeing `k8s.io/kubernetes/pkg/kubectl` unit test failures on cherry-picks..

**Comment by ixdy** [comment]
@karlkfi yes, the `release-1.1` branch has this test and is similarly broken, so the fix will need to be cherry-picked there.

**Comment by karlkfi** [comment]
The crisis has been averted, but @hurf will probably want to create a new unit test to replace the deleted one

**Comment by fabioy** [comment]
Lowering pri. Leaving the bug open for now to see if anyone's interested in fixing the unit test. Though, given the other test coverages, I wonder if it's needed at all.

**Comment by hurf** [comment]
I'll take care of it once I'm back to office. Fixed days like AddDate(0,0,-90) would work fine.

**Comment by bgrant0607** [comment]
This just failed again. Maybe daylight savings change?
kubernetes-jenkins/pr-logs/pull/22901/kubernetes-pull-test-unit-integration/18293
```
	resource_printer_test.go:1265: On 2 days ago, expected '2d', but got '1d'
	resource_printer_test.go:1265: On months ago, expected '90d', but got '89d'
```

**Comment by bgrant0607** [comment]
Still uses Now:
```
		{"a while from now", translateTimestamp(unversioned.Time{Time: time.Now().Add(2.1e9)}), "<invalid>"},
		{"almost now", translateTimestamp(unversioned.Time{Time: time.Now().Add(1.9e9)}), "0s"},
		{"now", translateTimestamp(unversioned.Time{Time: time.Now()}), "0s"},
		{"unknown", translateTimestamp(unversioned.Time{}), "<unknown>"},
		{"30 seconds ago", translateTimestamp(unversioned.Time{Time: time.Now().Add(-3e10)}), "30s"},
		{"5 minutes ago", translateTimestamp(unversioned.Time{Time: time.Now().Add(-3e11)}), "5m"},
		{"an hour ago", translateTimestamp(unversioned.Time{Time: time.Now().Add(-6e12)}), "1h"},
		{"2 days ago", translateTimestamp(unversioned.Time{Time: time.Now().AddDate(0, 0, -2)}), "2d"},
		{"months ago", translateTimestamp(unversioned.Time{Time: time.Now().AddDate(0, 0, -90)}), "90d"},
		{"10 years ago", translateTimestamp(unversioned.Time{Time: time.Now().AddDate(-10, 0, 0)}), "10y"},
```

**Comment by bgrant0607** [comment]
Build seems to be broken due to this now

**Comment by liggitt** [comment]
Might be related to DST being in the interval, not sure

**Comment by liggitt** [comment]
Fix is in #22929
